src/peer_digest.cc

```diff
@@ -699,23 +699,26 @@ peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const
     PeerDigest *pd = NULL;
     const char *reason = NULL;  /* reason for completion */
     const char *no_bug = NULL;  /* successful completion if set */
-    const int pdcb_valid = cbdataReferenceValid(fetch->pd);
-    const int pcb_valid = cbdataReferenceValid(fetch->pd->peer);
+    
+    // Here, the meaning of "_valid" differs from cbdata validity:
+    // Cbdata treats nil pointers as valid. This code does not.
+    const auto pdcb_valid = fetch->pd && cbdataReferenceValid(fetch->pd);
+    const auto pcb_valid = pdcb_valid && fetch->pd->peer && cbdataReferenceValid(fetch->pd->peer);
```

Comment:
This doesn't look right to me. It checks the value of peer against nullptr before checking the validity of that value.

FTR: these members should ideally be CbcPointer type and we can just rely on that API get() accessor.

Reply:
> This doesn't look right to me. It checks the value of peer against nullptr before checking the validity of that value.

I do not understand why you think that checking x for nil before calling cbdataReferenceValid(x) is not right. That sequence is (or should be) standard practice when one wants to dereference x. cbdataReferenceValid(nullptr) returns a truthy value so calling that function without checking for nil is not enough to dereference the pointer. This is one of the cbdata API problems that we cannot fix in this PR.